frontend/packages/ceph-storage-plugin/integration-tests/views/upgrade.view.ts

```diff
@@ -0,0 +1,77 @@
+import { execSync } from 'child_process';
+import { click } from '@console/shared/src/test-utils/utils';
+import * as crudView from '@console/internal-integration-tests/views/crud.view';
+import { goToInstalledOperators, ocsOperator } from './installFlow.view';
+import { selectItemFromDropdown } from './pvc.view';
+import { MINUTE, NS } from '../utils/consts';
+
+export const image43Command =
+  'oc get -n openshift-marketplace catalogSource ocs-catalogsource -o json|sed \'s/ocs-olm-operator:.*"/ocs-olm-operator:latest-stable-4.3.0"/g\'|oc apply -f -';
+export const image44Command =
+  'oc get -n openshift-marketplace catalogSource ocs-catalogsource -o json|sed \'s/ocs-olm-operator:.*"/ocs-olm-operator:latest-stable-4.4.0"/g\'|oc apply -f -';
+export const image45Command =
+  'oc get -n openshift-marketplace catalogSource ocs-catalogsource -o json|sed \'s/ocs-olm-operator:.*"/ocs-olm-operator:latest-stable-4.5.0"/g\'|oc apply -f -';
+
+// Subscription tab
+export const channelChangeButton = $('button[class="pf-c-button pf-m-link pf-m-inline"]');
+export const channel42 = $('input[value="stable-4\\.2"]');
+export const channel43 = $('input[value="stable-4\\.3"]');
+export const channel44 = $('input[value="stable-4\\.4"]');
+const saveChange = $('#confirm-action');
+// OCS Operator view
+export const subscription = element(by.partialLinkText('Subscription'));
+export const storageCluster = element(by.partialLinkText('Storage Cluster'));
+export const namespaceDropdown = $('[data-test-id="namespace-bar-dropdown"] div div button');
+
+export async function goToOCSOperator() {
+  await goToInstalledOperators();
+  await selectItemFromDropdown(NS, namespaceDropdown);
+  await browser.wait(until.and(crudView.untilNoLoadersPresent));
+  await click(ocsOperator);
+  await browser.wait(until.and(crudView.untilNoLoadersPresent));
+}
+
+export async function isStorageClusterReady() {
+  const storageClusterInfo = await JSON.parse(
+    execSync(`kubectl get storagecluster ocs-storagecluster -o json -n ${NS} `).toString(),
+  );
+  const storageClusterState = storageClusterInfo.status.phase;
+  return storageClusterState === 'Ready';
+}
+
+export async function storageClusterVersion() {
+  const storageClusterInfo = await JSON.parse(
+    execSync(`kubectl get storagecluster ocs-storagecluster -o json -n ${NS} `).toString(),
+  );
+  return storageClusterInfo.spec.version;
+}
+
+export async function waitUntilStorageClusterReady() {
+  await browser.wait(
+    isStorageClusterReady(),
+    20 * MINUTE,
+    'Storage Cluster should reach Ready state in 20 minutes',
+  );
+}
+
+export async function changeCatalogSourceImage(newValue: string) {
+  const newImageCommand = `oc get -n openshift-marketplace catalogSource ocs-catalogsource -o json|sed 's/ocs-olm-operator:.*\\"/ocs-olm-operator:${newValue}\\"/g'|oc apply -f -`;
+  await execSync(newImageCommand);
+  await browser.sleep(5 * MINUTE);
```

Comment:
Is there anything else we can wait on?

Reply:
I then wait for storage cluster to be in Ready state, but I added this sleep to make sure that waiting doesn't stop too early before the changes are actually applied.

Reply:
Can we write a function that checks whether the CR is updated to the required catalogSource. Rather than waiting for 5 minutes for that to happen.
We can take it up in a follow up PR later.